src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3198,10 +3198,10 @@ void os::Linux::rebuild_cpu_to_node_map() {
     // Get cpus from the original node and map them to the closest node. If node
     // is a configured node (not a memory-less node), then original node and
     // closest node are the same.
-    if (numa_node_to_cpus(nindex_to_node()->at(i), cpu_map, cpu_map_size * sizeof(unsigned long)) != -1) {
-      for (size_t j = 0; j < cpu_map_valid_size; j++) {
+    if (numa_node_to_cpus(nindex_to_node()->at(i), cpu_map, cpu_map_size * (int)sizeof(unsigned long)) != -1) {
+      for (int j = 0; j < cpu_map_valid_size; j++) {
         if (cpu_map[j] != 0) {
-          for (size_t k = 0; k < BitsPerCLong; k++) {
+          for (int k = 0; k < BitsPerCLong; k++) {
```

Comment:
For changes to lines 3145 to 3204, wouldn't it be better to keep using size_t?

Reply:
For some reason I have an aversion to seeing a `size_t` loop index variable. These changes to `int` are a pleasant surprise. :)

Reply:
I figured if the value fits in an int and it's complaining about casting to an int, then I can just use int.

Reply:
I like these changes too.